### PR TITLE
Check presence and names of model variables

### PR DESCRIPTION
**Pull request by michaelosthege**
related to #9 this PR adds checks about the names of important variables in the model and resulting inferencedata.

### CODE REVIEW

File: covid/tests.py [hunk 1]
```diff
@@ -34,10 +36,12 @@ def test_build(self):
...
         assert isinstance(pmodel, pymc3.Model)
-        # TODO: enable after PR #1 was merged:
-        # assert "date" in pmodel.coords
-        # TODO: make more asserts about dates & coords being part of the model
-        # TODO: assert presence of key random variables
+        # important coordinates
+        assert "date" in pmodel.coords
+        assert "nonzero_date" in pmodel.coords
+        # important random variables
+        for varname in ['r_t', 'seed', 'infections', 'test_adjusted_positive', 'exposure', 'positive', 'alpha']:
```
Comment (canyon289) [inline covid/tests.py:43]: Nit. This test will fail on the first assert. If possible would accumulate all the failures in a list or something and raise them all at once. Not a huge deal though

File: covid/tests.py [hunk 2]
```diff
@@ -22,6 +22,8 @@ def test_process(self):
         # the last entry in the data is the day before `run_date`!
         assert df_processed.xs('NY').index[-1] < run_date
         assert df_processed.xs('NY').index[-1] == (run_date - pandas.DateOffset(1))
+        assert "positive" in df_processed.columns
+        assert "total" in df_processed.columns
 
 
 class TestGenerative:
```
Comment (canyon289) [inline covid/tests.py:29]: These tests are comprehensive in checking model structure is right. Would suggest for future effort including a test that fitting some "very obvious data" returns the "very obvious" results to check that model computation is probably correct

File: covid/tests.py (hunk 1 above)
Comment (michaelosthege) [reply covid/tests.py]: done

File: covid/tests.py (hunk 2 above)
Comment (michaelosthege) [reply covid/tests.py:29]: in general yes, but the model takes a long time to sample and we don't have a generated test dataset yet...

**Pull request merged**